addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -211,6 +211,22 @@ function factory(dependencies) {
             return clear();
         }
 
+        /**
+         * @private
+         * @returns {string|undefined}
+         */
+        _computeDisplaySize() {
+            if (!this.size) return clear();
+            const units = ['bytes', 'Kb', 'Mb', 'Gb'];
+            let value = this.size;
+            let i = 0;
+            while (parseInt(value, 10) >= 1000) {
+                value /= 1000;
+                i++;
+            }
+            return `${Number.isInteger(value) ? value : value.toFixed(1)} ${units[i]}`;
```

Comment:
Should handle localization properly, so should make use of `env._t()` and maybe certainly string interpolation with `_.str.sprintf`.